libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/binder/ActionButtonBinder.kt

```diff
@@ -29,21 +29,9 @@ internal class ActionButtonBinder(
 
         val binding = MapboxActionButtonsLayoutBinding.bind(value)
         return navigationListOf(
-            AudioGuidanceButtonComponent(
-                navigationViewContext.viewModel.audioGuidanceViewModel,
-                navigationViewContext.viewModel.navigationStateViewModel,
-                binding.soundButton,
-            ),
-            CameraModeButtonComponent(
-                navigationViewContext.viewModel.cameraViewModel,
-                navigationViewContext.viewModel.navigationStateViewModel,
-                binding.cameraModeButton
-            ),
-            RecenterButtonComponent(
-                navigationViewContext.viewModel.cameraViewModel,
-                navigationViewContext.viewModel.navigationStateViewModel,
-                binding.recenterButton
-            )
```

Comment:
I like how much simpler this is! We had this originally, but separated them with the hope that they would be more portable. For example, someone could use the LocationComponent without needing all the state

Here is another approach that I've been considering. That way it is a "module" that defines which dependency is provided to the component.

Rather than the component being responsible for deciding what it needs from the context

``` kotlin
return navigationListOf(
  module.audioGuidanceComponent(binding.soundButton),
  module.cameraModeButtonComponent(binding.cameraModeButton),
  module.recenterButtonComponent(binding.recenterButton),
)
```

``` kotlin
UIModule {
  fun recenterButtonComponent = RecenterButtonComponent(
     viewModel.cameraViewModel,
     viewModel.navigationStateViewModel,
     recenterButton
  )
...
```

Reply:
I was thinking the same but named ComponentFactory.

```kotlin
internal class ComponentFactory(
    private val context: DropInNavigationViewContext
) {
    fun locationComponent(mapView: MapView) = LocationComponent(context, mapView)

    fun routeLineComponent(mapView: MapView, lineOptions: MapboxRouteLineOptions) =
        RouteLineComponent(context, mapView, lineOptions)

    fun cameraComponent(mapView: MapView) = CameraComponent(context, mapView)

    fun mapMarkersComponent(mapView: MapView) = MapMarkersComponent(context, mapView)

    fun routeArrowComponent(mapView: MapView, arrowOptions: RouteArrowOptions) =
        RouteArrowComponent(mapView, arrowOptions)
}
```
usage:

```kotlin
internal class DropInNavigationViewContext(...) {
 ...
 fun componentFactory() = ComponentFactory(this)
}
```
```kotlin
internal class ActiveGuidanceMapBinder(
    private val context: DropInNavigationViewContext,
) : Binder<MapView> {

    override fun bind(mapView: MapView): MapboxNavigationObserver {
        context.componentFactory().apply {
            return navigationListOf(
                locationComponent(mapView),
                reloadOnChange(
                    context.mapStyleLoader.loadedMapStyle,
                    context.options.routeLineOptions
                ) { _, lineOptions ->
                    routeLineComponent(mapView, lineOptions)
                },
                cameraComponent(mapView),
                mapMarkersComponent(mapView),
                reloadOnChange(
                    context.mapStyleLoader.loadedMapStyle,
                    context.options.routeArrowOptions
                ) { _, arrowOptions ->
                    routeArrowComponent(mapView, arrowOptions)
                }
            )
        }
    }
}
```

Reply:
👍  a Factory is good. 

Other point here if we require the `LocationComponent` to have an instance of `DropInNavigationViewContext`. Then we will not be able to create `LocationComponent` outside of the drop in ui.

But with the preference to take full adoption of `DropInNavigationViewContext`. Am thinking it makes sense to rename `DropInNavigationViewContext` to something more permanent. `DropInContext` for example.

Reply:
Or, we can make `ComponentFactory` public and introduce a method somewhere that will return it pre-wired for external use.

Reply:
Yeah I agree that it would be good to make the ComponentFactory public. Unclear to me how we would expose the `ComponentFactory` without an instance of `DropInNavigationView` if it depends on `DropInNavigationViewContext` 🤔 

Right now you can create a `LocationComponent` but you first need to create a `LocationViewModel`. You do not need to create the entire `DropInNavigationView`